src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1066,7 +1065,6 @@ public ListBoxModel doFillSshKeysCredentialsIdItems(@QueryParameter String sshKe
                     .includeCurrentValue(sshKeysCredentialsId);
         }
 
-        @RequirePOST
```

Comment:
It looks like fine to remove this annotation here, the method is just returning whether the credential-id showed is an actual rsa private key. But better to confirm with some security person: @daniel-beck, @Wadeck, ...

Reply:
There's still a permission check.

The docs for this are at https://www.jenkins.io/doc/developer/security/form-validation/ -- please read and if anything specific is unclear, ask.

Reply:
getSshCredential retrieves credentials, but as the RequirePOST protects against CSRF, there is no issue. The method is already checking for ADMINISTER.

Side note, such methods should not have troubles to use POST for the check. You can change the https://github.com/jenkinsci/ec2-plugin/blob/8702cc4a89bd7c096bba07211e222cde6d931f5d/src/main/resources/hudson/plugins/ec2/AmazonEC2Cloud/config-entries.jelly#L39 (and similar for Eucalyptus) by adding `checkMethod="post"`